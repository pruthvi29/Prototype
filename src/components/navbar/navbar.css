 img {
    height: 50px;
    padding: 7px;
}
.primary-nav .custom-nav {
    display: flex;
    width: 100%;
    background: linear-gradient(180deg,#338593,#1d6b7f) 0 no-repeat,50% no-repeat #066!important;
}
.primary-nav .custom-nav li {
    flex: 1 1 auto;
    border: 1px solid #000;
    border-right: none;
    text-align: center;
    border-bottom: 0px;
}
.primary-nav .custom-nav li a {
    color: #fff !important;
    padding: 10px;
    font-weight: bold;
    font-size: 16px;
}
 .navbar {
    margin-bottom: 0px !important;
    min-height: 30px !important;
    border: none !important;
}
 .secondary-nav .custom-nav {
    display: flex;
    width: 100%;
    background-color:#d8e1e8;
    min-height: 30px;
}
.secondary-nav .custom-nav li {
    flex: 1 1 auto;
    border: 1px solid #000;
    border-right: none;
    text-align: center;
}
.secondary-nav .custom-nav li a {
    color: #066 !important;
    padding: 2px;
    font-weight: bold;
    font-size: 14px;
}
.secondary-nav .custom-nav .active a, .custom-nav .active {
    color: #fff !important;
    background-color: #000 !important;
}
 .searchsection {
    margin: 0 15%;
    text-align: left;
    padding-top: 30px;
}
 input {
    width: 100%;
    border: 1px solid #ccc;
    border-radius: 5px;
    height: 35px;
    padding-left: 10px;
}
 .searchbutton {
    text-align: center;
    padding: 20px;
}
 button {
    background: linear-gradient(180deg,#338593,#1d6b7f) 0 no-repeat,50% no-repeat #066!important;
    color: #fff;
    border: none;
    border-radius: 3px;
    width: 70px;
    height: 30px;
    font-weight: bold;
}