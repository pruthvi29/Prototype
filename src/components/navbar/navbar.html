<nav class="navbar navbar-default">
  <ul class="nav navbar-nav custom-nav">
    <li v-for="item in items">
      <router-link v-bind:to="item.link">{{ item.name }}</router-link>
    </li>
  </ul>
</nav>