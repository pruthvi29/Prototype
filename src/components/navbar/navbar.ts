import Vue from 'vue';
import axios from 'axios';
import { Component, Watch } from 'vue-property-decorator';
import './navbar.css';
@Component({
  template: require('./navbar.html'),
  props: {
    value: null,
    items: {
      type: Array,
      required: true
    }
  }
})
export class Navbar extends Vue {

}