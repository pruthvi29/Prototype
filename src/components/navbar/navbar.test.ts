import Vue from 'vue';
import VueRouter from 'vue-router';
//import Component from 'vue-class-component';
import { Component, Watch } from 'vue-property-decorator';
import './navbar.css'
import { ComponentTest } from '../../util/component-test';
import { Navbar } from './navbar';

@Component({
  template: require('./navbar.html'),
  props: {
    value: null,
    items: {
      type: Array,
      required: true,
      default: function () { return [] }
    }
  }
})
class MockNavbarComponent extends Navbar {
  constructor() {
    super();
  }
}

describe('Navbar component', () => {
  let directiveTest: ComponentTest;
  let router: VueRouter;
  let items:[{'name': 'Home', 'link': '#'},{'name': 'Applications', 'link': '#'}];

  beforeEach(() => {
    Vue.use(VueRouter);
    directiveTest = new ComponentTest('<div><navbar :items='+items+'></navbar></div>', { 'Navbar': MockNavbarComponent });
  });


  it('should render correct contents', async () => {
    directiveTest.createComponent();
    await directiveTest.execute((vm) => { // ensure Vue has bootstrapped/run change detection
      expect(vm.$el).toMatchSnapshot();
    });
  });
});
