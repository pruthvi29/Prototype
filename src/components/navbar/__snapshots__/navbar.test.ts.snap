// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should render correct contents 1`] = `
<div>
  <nav
    class="navbar navbar-default"
  >
    <ul
      class="nav navbar-nav custom-nav"
    />
  </nav>
</div>
`;
